system/API/ResponseTrait.php

```diff
@@ -328,44 +328,31 @@ protected function format($data = null)
 		// If the data is a string, there's not much we can do to it...
 		if (is_string($data))
 		{
-			$this->setContentType('text/html');
+			// The content type should be text/... and not application/...
+			$contentType = $this->response->getHeaderLine('Content-Type');
+			$contentType = str_replace('application/json', 'text/html', $contentType);
+			$contentType = str_replace('application/', 'text/', $contentType);
```

Comment:
This line seems dangerous to me. While it's most likely that an API will respond with JSON or XML formatting, we can't anticipate the future and arbitrarily changing any content type that is `application/` to `text/` seems like we could be mucking with something without intending to. The most immediate concern I could think of is if someone is sending back a PDF which has a mime-type of `application/pdf`.